### DISCUSSION TITLE
Use conformance inside a monorepo with Next.js apps

### DISCUSSION THREAD

**Discussion by antoinelin** [vercel/next.js#26182] [Help]
Hello 🖖🏻 

Any advice to use Next.js ESLint on a monorepo with multiple apps ?
ATM I have issue to tells Next.js to lint files from my Nx Workspace apps and libs (it tries to lint from `/pages` or `/src/pages` instead of `apps/myapp/pages`)

Thanks 💪🏻

**Comment by tw0517tw** [comment]
Having similar issue here. We put out web app in a monorepo where we installed our `eslint` at root dir of the project and having the front-end thing like `react`, `next` installed in `client/`. What's the best practice to do `next lint` in such structure?

**Reply by tw0517tw** [reply]
OK I found a solution for our project.

- install `@next/eslint-plugin-next` in project root
- modify `client/.eslintrc.js` and 
  - add `plugin:@next/next/recommended` to `extends`
    - https://nextjs.org/docs/basic-features/eslint#migrating-existing-config
  - add `'@next/next/no-html-link-for-pages': ['warn', 'client/pages']` in `rules` because we run `eslint` from root so the dir option should be relative to project root
    - https://nextjs.org/docs/messages/no-html-link-for-pages
- add `eslint: { ignoreDuringBuilds: true }` to `client/next.config.js` since we run `eslint` from project root and has integrated in our CI process
  - https://nextjs.org/docs/api-reference/next.config.js/ignoring-eslint
- run `eslint` commands from project root as before

**Reply by antoinelin** [reply]
Ok nice ! I'm gonna try this, thanks! Does it provides well Next.js error during `lint`?

**Reply by tw0517tw** [reply]
![圖片](https://user-images.githubusercontent.com/1003146/122752840-0426b280-d2c4-11eb-96c0-fe16b4fad685.png)
It shows some warning from next plugin after I did above config. But I didn't check if every rules are working well.

**Reply by housseindjirdeh** [reply]
Thanks for the detailed steps @tw0517tw. Curious to hear if anything can be improved from the Next.js side?

**Reply by tw0517tw** [reply]
Hi @housseindjirdeh 

I think the document can be improved to mention more about integrating existed eslint config and execution. For example talk more about what does `next lint` do and what is the difference if we are already using something like `eslint .` for linting in [migrating existing config document](https://nextjs.org/docs/basic-features/eslint#migrating-existing-config). 

It would also be great if `next` can detect eslint config from parent folder if the eslint config in current folder is not `root: true`, then auto or prompt for user to set `ignoreDuringBuilds: true` in `next.config.js` and show a link to migration guide.

**Comment by Vadorequest** [comment]
Had a similar issue because my files are in `src` folder.

`next lint --dir src` fixed it for me.

So, `next lint --dir apps/myapp/pages` will do the job for you.

**Reply by tw0517tw** [reply]
I'm not sure if this will be the best practice. To run `next lint` at root folder means that we should install `next` as a dependency in project root. But we only has `next` installed inside `client/` folder right now.

**Reply by Vadorequest** [reply]
Yeah, I didn't really think about your path example. Looks like doing `next lint --dir myapp/pages` is what you should do. And if you need to run `lint` from root, then simply use a command that'll go inside `apps/myapp` and run lint from there.
